pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -323,19 +332,34 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
 
         String producerName = publishContext.getProducerName();
         long sequenceId = publishContext.getSequenceId();
+        // The process of the Producer sending chunk messages is continuous, and all chunks of a message use the same
```

Comment:
General Question: how does pulsar broker guarantee messages are received in the order? 

In other words, how do we guarantee that this `isDuplicate` is called in the consecutive order:e.g. m1-c1, m1-c2, m2-c1, m2-c2?

Do we have any documentation about it? I assume we rely on the netty tcp connection layer for this Channel.write, but I would like to confirm this.

Reply:
> I assume we rely on the netty tcp connection layer for this Channel.write, but I would like to confirm this.

IMO, it's that.